frontend/SplitPane.js

```diff
@@ -39,30 +44,45 @@ class SplitPane extends React.Component {
     super(props);
     this.state = {
       moving: false,
-      width: props.initialWidth,
+      width: (props.isVertical) ? '100%' : props.initialWidth,
+      height: (!props.isVertical) ? '100%' : props.initialHeight,
     };
   }
 
-  onMove(x: number) {
+  componentWillReceiveProps(nextProps: Props) {
+    this.setState({
```

Comment:
[It is dangerous to depend on `this.state` when calling `setState`](https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

And I think you don't need to. Maybe you can do something like this?

```js
if (nextProps.isVertical && !this.props.isVertical) {
  this.setState({width: '100%'});
} else if (!nextProps.isVertical && this.props.isVertical) {
  this.setState({height: '100%'});
}
```